Services/Services.Domain/Repositories.Abstractions/IDoctorRepository.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Services.Domain.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Services.Domain.Repositories.Abstractions
+{
+    public interface IDoctorRepository : IGenericRepository<Doctor>
+    {
+        Task<IEnumerable<Doctor>> GetActiveDoctorsWithSpecializationAsync(Guid specializationId, CancellationToken cancellationToken);
```

Comment:
Rename please, because you will filter by specializationId, this method should be named as "GetActiveDoctorsBySpecializationIdAsync"

Reply:
Fixed